### PR TITLE
DOC: deduplicate dtype basic types (2)

### LINKED ISSUE #17486: Deduplicate dtype documentation
## Documentation

We still have a fair bit of overlap between these two pages:

* `user/basics.types.rst`: https://numpy.org/devdocs/user/basics.types.html - table of scalar type and type aliases
* `reference/arrays.scalars.rst`: https://numpy.org/devdocs/reference/arrays.scalars.html - autodoc listings of scalar types and type aliases

It's tempting to remove the table on the first page, and tell users to look at the docs on the second.

If nothing else, it would be good to describe to the reader how the pages relate to each other - right now the former links to the latter only via the (currently broken) class names. A "see also" link somewhere would make the connection a little easier to understand.

### PR DESCRIPTION
**Pull request by tech-gian**
This is the continuation of PR #17505 . @melissawm @charris I think I got it right this time. I hope everything looks fine and this PR to be merged in the original repo!

Closes #17486.

### CODE REVIEW

File: doc/source/user/basics.types.rst [hunk 1]
```diff
@@ -96,70 +96,17 @@ The primitive types supported are tied closely to those in C:
...
-      - Unsigned integer (0 to 65535)
+In the site above, are listed some types of the table above such as `byte`, 
+`short`, `intc` and `int_`. The corresponding unsigned and signed types
+of these are written too.
 
-    * - `numpy.uint32`
-      - ``uint32_t``
-      - Unsigned integer (0 to 4294967295)
+'float' and 'complex' aliases are listed, as well. The functionality
+for each of these types are descripted in the table above, as long as
```
Comment (tylerjereddy) [inline doc/source/user/basics.types.rst:107]: `are descripted` -> `is described`?

`as long as` -> `as well as`?

Comment (tech-gian) [reply doc/source/user/basics.types.rst]: Fixed. Very nice comments. Thank you! Is it ready now for merging?

File: doc/source/user/basics.types.rst [hunk 2]
```diff
@@ -96,70 +96,17 @@ The primitive types supported are tied closely to those in C:
 
 
 Since many of these have platform-dependent definitions, a set of fixed-size
-aliases are provided:
+aliases are provided to this site: :doc:`../reference/arrays.scalars`
+under the "Sized aliases" section.
```
Comment (melissawm) [inline doc/source/user/basics.types.rst:100]: I suggest adding a proper link to this section. This means the current text will read 
```suggestion
aliases are provided (See :ref:`sized-aliases`).
```

You will also need to edit the file `doc/source/reference/arrays.scalars.rst` and just before the "Sized aliases" section you should add the following:

`.. _sized-aliases:`

FInally, you should always build the documentation locally to check that these references and links are working. If you have trouble doing that please mention it here and I can help.

File: doc/source/user/basics.types.rst [hunk 3]
```diff
@@ -96,70 +96,17 @@ The primitive types supported are tied closely to those in C:
...
+In the site above, are listed some types of the table above such as `byte`, 
+`short`, `intc` and `int_`. The corresponding unsigned and signed types
+of these are written too.
 
-    * - `numpy.uint32`
-      - ``uint32_t``
-      - Unsigned integer (0 to 4294967295)
+'float' and 'complex' aliases are listed, as well. The functionality
+for each of these types is descripted in the table above, as well as
+examples with similarities with ``C types``.
```
Comment (melissawm) [inline doc/source/user/basics.types.rst:108]: I don't think this text is necessary, as we've already added a link to the section where this is detailed.

File: doc/source/user/basics.types.rst [hunk 4]
```diff
@@ -171,7 +118,7 @@ having unique characteristics.  Once you have imported NumPy using
 
 the dtypes are available as ``np.bool_``, ``np.float32``, etc.
 
-Advanced types, not listed in the table above, are explored in
+Advanced types, not listed in the site above, are explored in
```
Comment (melissawm) [inline doc/source/user/basics.types.rst:121]:
```suggestion
Advanced types, not listed above, are explored in
```

I think the use of the word "site" does not apply here, but I could be wrong :)

**Review by melissawm — CHANGES_REQUESTED** [review]
Thanks, @tech-gian ! I believe we still might have trouble as you have renamed your main branch but still did your PR against it. You should always create a `feature branch` for your PRs - but maybe we can make this work here after you do the textual changes. Cheers!

File: doc/source/user/basics.types.rst (hunk 4 above)
Comment (tech-gian) [reply doc/source/user/basics.types.rst]: Fixed.

File: doc/source/user/basics.types.rst (hunk 3 above)
Comment (tech-gian) [reply doc/source/user/basics.types.rst]: I read it again and I think that too.

File: doc/source/user/basics.types.rst (hunk 2 above)
Comment (tech-gian) [reply doc/source/user/basics.types.rst]: Fixed.

**Comment by tech-gian** [comment]
I created 2 new commits with all the changes you suggested. Is there anything I should do or I just wait for you to accept my changes to the original repository?

**Review by melissawm — APPROVED** [review]
LGTM, thanks!

**Comment by charris** [comment]
Thanks @tech-gian .

**Pull request merged**